pytorch_lightning/loops/utilities.py

```diff
@@ -0,0 +1,104 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Mapping, Optional, Tuple
+
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.utilities.apply_func import apply_to_collection
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.finite_checks import detect_nan_parameters
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+
+def check_finite(model: "pl.LightningModule", loss: torch.Tensor) -> None:
+    """Checks for finite parameters and loss values.
+
+    Args:
+        model: a reference to the ``LightningModule``
+        loss: the loss value to check to be finite
+    """
+    if not torch.isfinite(loss).all():
+        raise ValueError(f"The loss returned in `training_step` is {loss}.")
+    detect_nan_parameters(model)
+
+
+def check_training_step_output(model: "pl.LightningModule", training_step_output: STEP_OUTPUT) -> None:
```

Comment:
```suggestion
def _check_training_step_output(model: "pl.LightningModule", training_step_output: STEP_OUTPUT) -> None:
```
lets keep it protected